BitwardenShared/UI/Platform/Application/AppProcessor.swift

```diff
@@ -84,6 +84,7 @@ public class AppProcessor {
                 startEventTimer()
                 await checkIfExtensionSwitchedAccounts()
                 await checkAccountsForTimeout()
+                await coordinator?.handleEvent(.didStart)
```

Comment:
⚠️ We don't want to call this every time the app is foregrounded as it will reset the UI back to the vault list. I think we need to check the following and scope it to the specific scenario we're looking for:

- If not in the extension
- If vault is locked
- If the user uses never lock
- If the user hasn't manually locked their vault (because it was unlocked in the extension). `!stateService.getManuallyLockedAccount()`

If the above are true, we should be safe to unlock the vault. I don't really like repurposing the `didStart` event for this since the app isn't starting up. `AuthEvent.accountBecameActive` is probably the most fitting event, we should just need a new `AppEvent` that maps to that.